### PR TITLE
Port the robust vertex access tests from GLSL to WGSL

### PR DESCRIPTION
**Pull request by Kangz**
Currently blocked on https://bugs.chromium.org/p/tint/issues/detail?id=355

### CODE REVIEW

**Comment by Kangz** [comment]
PTAL https://github.com/gpuweb/cts/compare/glsl-dependent..Kangz:wgsl_robust_vertex_access#diff-b27f36ba13f8198439e1b38dd725bb06ba69849f1df296dbfbc51535b3e20644

I had to fix a couple validation errors (in particular for mapped buffers). In this current state this CL passes on D3D12 (except for the baseVertex tests) and correctly crashes my mac ^^.

**Comment by Kangz** [comment]
PTAL

File: src/webgpu/shader/execution/robust_access_vertex.spec.ts [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+                   VertexIndex < ${vertexIndexOffset + numVertices}u);
+
+              if (attributesInBounds && (${!p.indexed} || indexInBounds)) {
+                # Success case, move the vertex out of the viewport
+                Position = vec4<f32>(-1.0, 0.0, 0.0, 1.0);
+              } else {
+                # Failure case, move the vertex inside the viewport
+                Position = vec4<f32>(0.0, 0.0, 0.0, 1.0);
+              }
+              return;
```
Comment (austinEng) [inline src/webgpu/shader/execution/robust_access_vertex.spec.ts:385]: nit: explicit return should no longer be necessary

File: src/webgpu/shader/execution/robust_access_vertex.spec.ts [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+              fragColor = vec4<f32>(1.0, 0.0, 0.0, 1.0);
+              return;
+            }`,
+        }),
+        entryPoint: 'main',
+      },
+      primitiveTopology: 'point-list',
+      colorStates: [{ format: 'rgba8unorm' }],
+      vertexState: {
+        indexFormat: 'uint16',
```
Comment (austinEng) [inline src/webgpu/shader/execution/robust_access_vertex.spec.ts:404]: pre-existing: shouldn't need indexFormat

**Review by austinEng — APPROVED** [review]
LGTM

**Pull request merged**